rbd/watchers_mimic_test.go

```diff
@@ -0,0 +1,160 @@
+// +build !luminous,!mimic
+//
+// Ceph Nautilus is the first version that supports watchers through librbd.
+
+package rbd
+
+import (
+	"testing"
+
+	"github.com/ceph/go-ceph/rados"
+	"github.com/stretchr/testify/assert"
+)
+
+func TestListWatchers(t *testing.T) {
+	conn, _ := rados.NewConn()
+	conn.ReadDefaultConfigFile()
+	conn.Connect()
+
+	poolname := GetUUID()
+	err := conn.MakePool(poolname)
+	assert.NoError(t, err)
+
+	ioctx, err := conn.OpenIOContext(poolname)
+	assert.NoError(t, err)
+
+	name := GetUUID()
+	image, err := Create(ioctx, name, 1<<22, 22)
+	assert.NoError(t, err)
+
+	// will not work on an image that is not open
+	_, err = image.ListWatchers()
+	assert.Equal(t, ErrImageNotOpen, err)
+
+	err = image.Open(true) // readonly
```

Comment:
Could I pester you to isolate sections of the test that are (mostly) independent into t.Run(...) calls? It really highlights which conditions are being tested, isolates vars, and improves error reporting on test failures (IMO). I wont hold the PR over it but it would be nice if you'd consider adding that.

Reply:
Sure, I can do that. While looking at this, I notice that it also uses the old style of create/opening images. Will correct that too.